lib/open_api_spex/operation.ex

```diff
@@ -120,15 +120,36 @@ defmodule OpenApiSpex.Operation do
   @doc """
   Cast params to the types defined by the schemas of the operation parameters and requestBody
   """
-  @spec cast(Operation.t, map, String.t | nil, %{String.t => Schema.t}) :: {:ok, map} | {:error, String.t}
-  def cast(operation = %Operation{}, params = %{}, content_type, schemas) do
-    parameters = Enum.filter(operation.parameters || [], fn p -> Map.has_key?(params, Atom.to_string(p.name)) end)
-    with {:ok, parameter_values} <- cast_parameters(parameters, params, schemas),
-         {:ok, body} <- cast_request_body(operation.requestBody, params, content_type, schemas) do
+  @spec cast(Operation.t, Conn.t, String.t | nil, %{String.t => Schema.t}) :: {:ok, map} | {:error, String.t}
+  def cast(operation = %Operation{}, conn = %{}, content_type, schemas) do
+    parameters = Enum.filter(operation.parameters || [], fn p -> Map.has_key?(conn.params, Atom.to_string(p.name)) end)
+    with :ok <- validate_parameters(conn, operation.parameters),
+         {:ok, parameter_values} <- cast_parameters(parameters, conn.params, schemas),
+         {:ok, body} <- cast_request_body(operation.requestBody, conn.params, content_type, schemas) do
       {:ok, Map.merge(parameter_values, body)}
     end
   end
 
+  defp validate_parameters(conn, [_] = defined_parameters) do
```

Comment:
this `[_]` pattern matches a list with exactly 1 element, is that what you wanted it to do?